src/uct/ib/base/ib_md.c

```diff
@@ -328,8 +329,12 @@ static ucs_status_t uct_ib_md_reg_mr(uct_ib_md_t *md, void *address,
 
         mr = UCS_PROFILE_CALL(ibv_exp_reg_mr, &in);
         if (mr == NULL) {
-            ucs_log(level, "ibv_exp_reg_mr(address=%p, length=%zu, exp_access=0x%lx) failed: %m",
-                    in.addr, in.length, in.exp_access);
+            ucs_log(level, "ibv_exp_reg_mr(address=%p, length=%zu, exp_access=0x%lx) failed. "
+                    "The max locked memory limit (ulimit -l) on the system is set to %s. "
```

Comment:
1. there is code duplication
2. pls print the message about ulimit only if the current limit is not unlimited (a finite number or unknown)